### PR TITLE
add docker services

### LINKED ISSUE #14: Setup docker structure
create 3 docker services:
1 - BD Postgress
2 - Backend
3 - nginx

### PR DESCRIPTION
**Pull request by lfjnascimento**
## Description

Add docker-compose.yml creating 3 services:
- database service using postgresql;
- backend service for our django app;
- proxy service with nginx.

## Related Issues
- Closes #14 

## How to test it

Create secret files: 
```sh
mkdir -p backend/runtime/secrets
uuidgen > backend/runtime/secrets/postgres_password_secret
```
Startup the services with
 ```sh
 docker compose up --build -d
 ```

Make a request to the proxy, it should return the default Django content as we don't have any custom endpoints for now
 ```sh
 curl http://localhost:80
 ```

### CODE REVIEW

File: backend/poetry.lock
Comment (mari1912) [inline backend/poetry.lock:1]: why we are adding this file? the poetry.toml ahould be enough

File: backend/pyproject.toml [hunk 1]
```diff
@@ -9,6 +9,8 @@ readme = "README.md"
 python = "^3.10"
 django = "^5.0.6"
 djangorestframework = "^3.15.2"
+gunicorn = "^22.0.0"
```
Comment (mari1912) [inline backend/pyproject.toml:12]: why are we adding this?

File: backend/runtime/secrets/postgres_password_secret
Comment (mari1912) [inline backend/runtime/secrets/postgres_password_secret:1]: why we don't use a .env file?

**Review by mari1912 — COMMENTED** [review]
The build is failing

File: backend/runtime/secrets/postgres_password_secret
Comment (lfjnascimento) [reply backend/runtime/secrets/postgres_password_secret:1]: Oops, this shouldn't be here. This is a docker solution for secrets, you can read more about it in [this doc](https://docs.docker.com/compose/use-secrets/)

File: backend/pyproject.toml (hunk 1 above)
Comment (lfjnascimento) [reply backend/pyproject.toml:12]: we will use `gunicorn` as a server for the Django application

Comment (mari1912) [reply backend/pyproject.toml:12]: but the django already have a server I think, it's just run `poetry run python3 manage.py run server`

File: backend/utils/docker/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      \"connect_timeout\": ${DB_DEFAULT_TIMEOUT:=2}
+    }
+}"
+
+if [ "$SETUP_DJANGO" = 1 ]; then
+    ./manage.py migrate --noinput
+    exit 0
+fi
+
+exec gunicorn kernelCI.wsgi:application "$@"
```
Comment (mari1912) [inline backend/utils/docker/entrypoint.sh:49]: we should run django here, not gunicorn poetry run python3 manage.py run server`

File: backend/pyproject.toml (hunk 1 above)
Comment (lfjnascimento) [reply backend/pyproject.toml:12]: Django server is a tool to make the development straight forward, it is not a "real server"

**Review by mari1912 — APPROVED** [review]

**Pull request merged**